### ISSUE TITLE
[Linux][RPi]Allow GLFW to use GLES instead of core OpenGL?

### ISSUE THREAD

**Issue by awe444** [Rubberduckycooly/RSDKv5-Decompilation#9]
Thank you for releasing this amazing decompilation!

I’ve built the binaries on 32-bit ARM Linux (Raspberry Pi OS Bullseye). When I try to run the game it fails, with log.txt showing that it successfully loaded a bunch of asset files from Data.rsdk but ultimately errored with:

`ERROR: [GLFW] window creation failed`

Now, I definitely have a working install of GLFW on my setup— I even built some of the [example GLFW demos](https://github.com/glfw/glfw/tree/master/examples) to convince myself of this. What I suspect is happening here is that RSDKv5 expects a core OpenGL version >= 3.0, but my low-end setup here only has OpenGL 2.1 and OpenGL ES 3.1. These lines seem to be the culprits:

https://github.com/Rubberduckycooly/RSDKv5-Decompilation/blob/82d1e1c0a8fb3e60e89784a3514504317a88e0b3/RSDKv5/RSDK/Graphics/GLFW/GLFWRenderDevice.cpp#L1

https://github.com/Rubberduckycooly/RSDKv5-Decompilation/blob/82d1e1c0a8fb3e60e89784a3514504317a88e0b3/RSDKv5/RSDK/Graphics/GLFW/GLFWRenderDevice.cpp#L56

I’ll try some workarounds such as using environment variables to enable Mesa compatibility mode when running, but it would be great if GLFW could be configured (e.g. at build time with preprocessor definitions) to use GLES on Linux if a suitably-modern core GL version isn’t available.

Thank you again

EDIT/UPDATE: Confirmed that it works if I override the GL version like this when running it;
```
MESA_GL_VERSION_OVERRIDE=3.3 MESA_GLSL_VERSION_OVERRIDE=330 ./RSDKv5
```
From what little of the game I tested so far it runs at full speed without glitches, so this is definitely a workaround. It would however still be nice to have native GLES support in Linux.

**Comment by awe444** [comment]
Looks great over the Pi’s composite video output by the way:
![5FB7348C-507E-47AF-B191-555827FA27A2](https://user-images.githubusercontent.com/97898022/185792562-855f09f6-6ec6-4657-8c52-56801898ca2d.jpeg)

**Comment by MGRich** [comment]
Will see about add a few defines for changing these on compile time.

**Comment by MGRich** [comment]
Using the newest commits, do they allow it to build and run natively for the Pi?

**Comment by awe444** [comment]
> Using the newest commits, do they allow it to build and run natively for the Pi? GLFW now targets GL 2.0 by default (no specified GLES compatibility yet(?))

So, I just tried using the new OGL subsystem on RPi4. It builds, but crashes before anything comes on screen, with log.txt containing the error `ERROR: [GLFW] window creation failed`.

HOWEVER, having improved my understanding of how to properly build the various subsytems since creating the OP, this issue can probably be closed as irrelevant/inconsequential because **_Pi users can (and should really prefer to) use the SDL2 backend_**. To be clear to everyone reading, this means adding a `-DRSDK_USE_SDL2=1` build flag and building with `make SUBSYSTEM=SDL2`.

Ideally the Pi user will also have, as a prerequisite:

- natively built their own Mesa library with GLES2 + VC4 support
- natively built their own SDL2 library with GLES2 + KMSDRM support
- define `SDL_VIDEODRIVER=kmsdrm` environment variable prior to running the executable

I've done the above and the game works great on Pi 4 with SDL2, without any complications related to GL versions. I have **not** however tested it using pre-built mesa and SDL2 libraries from the raspian apt repository. 

Also worth mentioning that the (potentially) SDL2-related error I had reported in https://github.com/Rubberduckycooly/RSDKv5-Decompilation/issues/124 does not show up when using SDL2 on RPi.
